### PR TITLE
INT-4301: Make CircuitBreakerOpenException as ME

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4301

To keep the context of the `failedMessage` and avoid extra wrapping
to the `MessagingException` make
`CircuitBreakerOpenException extends MessagingException` directly

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RequestHandlerCircuitBreakerAdvice.java [hunk 1]
```diff
@@ -99,12 +102,12 @@ private AtomicInteger getFailures() {
...
 	 * An exception thrown when the circuit breaker is in an open state.
 	 */
-	public static final class CircuitBreakerOpenException extends RuntimeException {
+	public static final class CircuitBreakerOpenException extends MessagingException {
 
 		private static final long serialVersionUID = 1L;
 
-		CircuitBreakerOpenException(String message) {
-			super(message);
+		public CircuitBreakerOpenException(Message<?> message, String description) {
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RequestHandlerCircuitBreakerAdvice.java]: Why the change to public? I wouldn't expect a user to need to instantiate one.

Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RequestHandlerCircuitBreakerAdvice.java]: Doh. That is artifact of the automatic IDEA refactoring. Won't mind if you fix on merge.

Thanks

**Comment by garyrussell** [comment]
Merged as 6829da995f164b6481cbf7c5a691cb9508c6d87a
